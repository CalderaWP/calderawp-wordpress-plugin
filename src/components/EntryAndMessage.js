import React, {Component, Fragment} from 'react';

import {MailDataView} from "./Message/MailDataView";
import {EntryHeaders} from "./EntryViewer/EntryHeaders";
import {SingleEntry} from "./EntryViewer/SingleEntry";
import {MailDataEdit} from "./Message/MailDataEdit";

export class EntryAndMessage extends Component {

	constructor(props){
		super(props);
		this.state = {
			mailData: props.mailData,
			showEdit: false,
		};
		this.setMailData = this.setMailData.bind(this)
		this.updateMessage = this.updateMessage.bind(this)
		this.setShowEdit = this.setShowEdit.bind(this)
	}

	setShowEdit(showEdit){
		this.setState({showEdit});
	}
	setMailData(mailData){
		this.setState({mailData});

	}

	updateMessage(message){
		const {mailData} = this.state;

		this.setMailData({
			...mailData,
			message
		});
	}

	render(){
		const {props,state} = this;
		const {
			entries,
			entry,
			formFields,
			formId,
			entryId,
		} = props;

		const {
			mailData,
			showEdit
		} = state;

		const closeMessage = props.closeMessage ? props.closeMessage : 'Close Editor';
		const openMessage = props.openMessage ? props.openMessage : 'Edit Message';
		return(
			<Fragment>
				<div>
					<div>
						<h2>Entry Data</h2>
						<EntryHeaders
							entries={entries}
						  	formFields={formFields}
						/>
						<SingleEntry
							entry={entry}
							entryId={entryId}
							formId={formId}
						/>
					</div>
					<div>
						{ mailData &&
						<Fragment>
							<h2>Primary Message</h2>
							<button
								className={'toggle-edit-view'}
								onClick={() =>this.setShowEdit(! showEdit )}
							>
								{showEdit ? (
									<Fragment>{closeMessage}</Fragment>

								) : (
									<Fragment>{openMessage}</Fragment>
								)}
							</button>
							{showEdit ? (
								<MailDataEdit
									mailData={mailData}
									onEditMessage={this.updateMessage}
									onChangeMailData={this.setMailData}
								/>

							) : (
								<MailDataView
									mailData={mailData}
								/>
							)}


						</Fragment>


						}
					</div>

				</div>

			</Fragment>
		)
	}
}
