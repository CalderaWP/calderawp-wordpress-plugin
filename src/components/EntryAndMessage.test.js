import React from "react";
import renderer from "react-test-renderer";
import "@babel/polyfill";
import {EntryAndMessage} from "./EntryAndMessage";
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

const mailData = {
	attachments: [],
	bcc: false,
	csv: false,
	from: "test@test.com",
	from_name: "Caldera Forms Notification",
	html: true,
	message: "<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">Mike</div>\n<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">Pollock</div>\n<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">jâ€¦",
	recipients:
		['roy@hiroy.club','alsoroy@hiroy.club'],
	replyto: "test@test.com",
	subject: "Contact Form 1"
};

const entries = require( '../__MOCK_DATA__/entries' );

const entry = {...entries[1],mailData};
describe( 'EntryAndMessage component' , () => {
	it( 'matches snapshot', () => {
		expect(renderer.create(
			<EntryAndMessage
				mailData={mailData}
				entries={entries}
				entryId={entry.id}
				entry={entry}
				formId={entry.form_id}
				formFields={entry.fields}
			/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'updates mailData in state',() => {
		const component = shallow(
			<EntryAndMessage
				mailData={mailData}
				entries={entries}
				entryId={entry.id}
				entry={entry}
				formId={entry.form_id}
				formFields={entry.fields}
			/>
		);
		const newData = {...mailData, subject: 'Noms Time!' };
		component.instance().setMailData(newData);
		expect( component.state( 'mailData') ).toEqual(newData);
	});

	it( 'updates message in state',() => {
		const component = shallow(
			<EntryAndMessage
				mailData={mailData}
				entries={entries}
				entryId={entry.id}
				entry={entry}
				formId={entry.form_id}
				formFields={entry.fields}
			/>
		);
		component.instance().updateMessage('Foo');
		expect( component.state( 'mailData').message ).toEqual('Foo');
	});

	it( 'Sets edit show',() => {
		const component = shallow(
			<EntryAndMessage
				mailData={mailData}
				entries={entries}
				entryId={entry.id}
				entry={entry}
				formId={entry.form_id}
				formFields={entry.fields}
			/>
		);
		component.instance().setShowEdit(true);
		expect( component.state( 'showEdit') ).toEqual(true);
		component.instance().setShowEdit(false);
		expect( component.state( 'showEdit') ).toEqual(false);
	});

	it( 'Updates button message', () => {
		const closeMessage = 'Close!';
		const openMessage= 'Open!';
		const component = shallow(
			<EntryAndMessage
				mailData={mailData}
				entries={entries}
				entryId={entry.id}
				entry={entry}
				formId={entry.form_id}
				formFields={entry.fields}
				closeMessage={closeMessage}
				openMessage={openMessage}
			/>
		);
		expect( component.find( '.toggle-edit-view').text() ).toEqual(openMessage);
		component.setState( {showEdit:true});
		expect( component.find( '.toggle-edit-view').text() ).toEqual(closeMessage);

	})
});
