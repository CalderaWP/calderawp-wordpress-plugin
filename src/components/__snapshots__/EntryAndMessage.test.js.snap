// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryAndMessage component matches snapshot 1`] = `
<div>
  <div>
    <h2>
      Entry Data
    </h2>
    <tr>
      <th
        scope="col"
      >
        ID
      </th>
      <th>
        first_name
      </th>
      <th>
        last_name
      </th>
      <th>
        email_address
      </th>
      <th>
        comments_questions
      </th>
    </tr>
    <tr>
      <th
        scope="row"
      >
        2
      </th>
      <td>
        Roy
      </td>
      <td>
        Sivan
      </td>
      <td>
        roy@hiroy.club
      </td>
      <td>
        fs
      </td>
    </tr>
  </div>
  <div>
    <h2>
      Primary Message
    </h2>
    <button
      className="toggle-edit-view"
      onClick={[Function]}
    >
      Edit Message
    </button>
    <div
      className="wp-block-columns has-2-columns"
    >
      <div
        className="wp-block-column"
      >
        <h3>
          Headers
        </h3>
        <ul>
          <li>
            <span>
              From
              :
            </span>
             
            test@test.com
          </li>
          <li>
            <span>
              From_name
              :
            </span>
             
            Caldera Forms Notification
          </li>
          <li>
            <span>
              Html
              :
            </span>
             
            HTML
          </li>
          <li>
            <span>
              Recipients
              :
            </span>
             
            roy@hiroy.club
            alsoroy@hiroy.club
          </li>
          <li>
            <span>
              Replyto
              :
            </span>
             
            test@test.com
          </li>
          <li>
            <span>
              Subject
              :
            </span>
             
            Contact Form 1
          </li>
        </ul>
      </div>
      <div
        className="wp-block-column"
      >
        <h3>
          Message
        </h3>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong></strong></p>
            <div style=\\"margin-bottom:20px;\\">Mike</div>
            <p><strong></strong></p>
            <div style=\\"margin-bottom:20px;\\">Pollock</div>
            <p><strong></strong></p>
            <div style=\\"margin-bottom:20px;\\">jâ€¦",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
